amaranth/vendor/gowin.py

```diff
@@ -221,6 +221,7 @@ def _device_name(self):
             set_option -verilog_std v2001 -print_all_synthesis_warning 1 -show_all_warn 1
             {{get_override("add_options")|default("# (add_options placeholder)")}}
             run all
+            file copy -force impl/pnr/project.fs {{name}}.fs
```

Comment:
Thanks, this is a really clever workaround!

Reply:
Thanks. Let's have TCL take care of this business.

Reply:
Ugh, project.fs is created without w permission, so can't easily be overwritten by a subsequent copy. Setting file attributes is not OS independent in TCL (throws 'bad option' if using invalid options for the current OS). Trying to instead delete destination before copy.

Reply:
Yes, deleting before copying, or moving instead of copying, should both work.